### PR TITLE
Prevents ID allocation beyond bitmap bounds

### PR DESCRIPTION
**Pull request by Fischer0522**
This PR adds boundary validation to `alloc_consecutive` to match the existing behavior in `alloc`, preventing out-of-range allocations in `IdAlloc` and `IdBitmap` that could cause panics in the underlying library.

### CODE REVIEW

File: kernel/src/fs/utils/id_bitmap.rs [hunk 1]
```diff
@@ -108,6 +108,11 @@ impl IdBitmap {
             return None;
         }
 
+        let end = self.first_available_id.checked_add(count)?;
+        if end > self.len {
+            return None;
```
Comment (tatetian) [inline kernel/src/fs/utils/id_bitmap.rs:113]: Could you add a unit test to show the exact panicking bug that this PR is trying to fix?

Comment (Fischer0522) [reply kernel/src/fs/utils/id_bitmap.rs:113]: This is a corner case that occurs when `first_available_id == len` and `count == 1`.

In this scenario, the allocation bypasses the existing boundary check:
```rust
            if curr_range.len() < count {
                return None;
            }
```
because `curr_range.len()` equals 1, satisfying the condition. However, the
resulting index is out of bounds, causing a panic in the underlying bitvec library:
```
thread 'test::test_bitmap_alloc_out_of_bounds' (1150802) panicked at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bitvec-1.0.1/src/slice.rs:1585:3:
index 16 out of range: Excluded(16)
```

**Comment by lrh2000** [comment]
@boterinas rerun

**Comment by lrh2000** [comment]
I am not sure why the rerun workflow still passes the lint CI... This PR should fail CI since https://github.com/asterinas/asterinas/pull/2959 has already been merged.

In my local environment, `make check` does fail:
```
error: redundant `test_` prefix in test function name
   --> ostd/libs/id-alloc/src/lib.rs:177:8
    |
177 |     fn test_bitmap_alloc_out_of_bounds() {
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing the `test_` prefix: `bitmap_alloc_out_of_bounds`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_test_prefix
    = note: `-D clippy::redundant-test-prefix` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::redundant_test_prefix)]`

error: could not compile `id-alloc` (lib test) due to 1 previous error
```

File: ostd/libs/id-alloc/src/lib.rs [hunk 2]
```diff
@@ -163,3 +168,22 @@ impl Debug for IdAlloc {
...
+    fn test_bitmap_alloc_out_of_bounds() {
+        let capacity = 16;
+        let mut bitmap = IdAlloc::with_capacity(capacity);
+
+        for _ in 0..capacity {
+            assert!(bitmap.alloc().is_some());
+        }
+
+        // Allocating one more ID should fail since the
+        // bitmap's first_available_id + count is out of bounds.
```
Comment (lrh2000) [inline ostd/libs/id-alloc/src/lib.rs]: By the way:
```suggestion
        // bitmap's `first_available_id` + `count` is out of bounds.
```

**Comment by tatetian** [comment]
> I am not sure why the rerun workflow still passes the lint CI... This PR should fail CI since #2959 has already been merged.

Here is my theory: it is the `.github/` directory of _this_ PR's branch that Github CI will respect, not the one in the main branch. After this PR is rebased to the main branch, the link will work.

File: kernel/src/fs/utils/id_bitmap.rs (hunk 1 above)
Comment (tatetian) [reply kernel/src/fs/utils/id_bitmap.rs:113]: OK. Thanks for explaining.

**Comment by Fischer0522** [comment]
> LGTM. Just remove the `test_` prefix and fix the nits.

All done.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!
